package.json

```diff
@@ -52,48 +52,51 @@
   "author": "Hibernating Rhinos",
   "license": "MIT",
   "devDependencies": {
-    "@types/bluebird": "^3.5.25",
-    "@types/mocha": "^5.0.0",
-    "@types/sinon": "^2.3.6",
-    "@types/ws": "^6.0.1",
+    "@types/bluebird": "^3.5.32",
+    "@types/md5": "^2.2.0",
+    "@types/mocha": "^7.0.2",
+    "@types/pluralize": "0.0.29",
+    "@types/sinon": "^9.0.4",
+    "@types/ws": "^7.2.5",
     "cross-os": "^1.3.0",
-    "glob": "^7.1.3",
+    "glob": "^7.1.6",
+    "http-proxy-agent": "^4.0.1",
     "lodash.orderby": "^4.6.0",
-    "mocha": "^5.2.0",
-    "nyc": "^14.1.0",
-    "sinon": "^7.2.3",
-    "source-map-support": "^0.5.9",
-    "tslint": "^5.12.0",
+    "mocha": "^8.0.1",
+    "nyc": "^15.1.0",
+    "sinon": "^9.0.2",
+    "source-map-support": "^0.5.19",
+    "ts-node": "^8.10.2",
+    "tslint": "^6.1.2",
     "tslint-eslint-rules": "^5.4.0",
-    "tslint-microsoft-contrib": "^6.0.0",
-    "typescript": "^3.2.2"
+    "tslint-microsoft-contrib": "^6.2.0",
+    "typescript": "^3.9.5"
   },
   "dependencies": {
-    "@types/pluralize": "0.0.27",
-    "@types/qs": "^6.5.1",
-    "@types/request": "^2.48.1",
+    "@types/node-fetch": "^2.5.7",
+    "@types/qs": "^6.9.3",
     "@types/semaphore": "^1.1.0",
-    "@types/uuid": "^2.0.29",
-    "@types/verror": "^1.10.3",
-    "bluebird": "^3.5.3",
-    "change-case": "^3.0.2",
-    "deprecate": "^1.1.0",
-    "md5-hex": "^2.0.0",
-    "moment": "^2.23.0",
-    "object.entries": "^1.0.4",
-    "object.values": "^1.0.4",
-    "pluralize": "^4.0.0",
-    "qs": "^6.6.0",
-    "readable-stream": "^3.1.0",
-    "request": "^2.88.0",
-    "safe-memory-cache": "^1.5.1",
+    "@types/uuid": "^8.0.0",
+    "@types/verror": "^1.10.4",
+    "bluebird": "^3.7.2",
+    "change-case": "^3.1.0",
+    "deprecate": "^1.1.1",
+    "md5": "^2.2.1",
+    "moment": "^2.27.0",
+    "node-fetch": "^2.6.0",
+    "object.entries": "^1.1.2",
+    "object.values": "^1.1.1",
+    "pluralize": "^8.0.0",
+    "qs": "^6.9.4",
+    "readable-stream": "^3.6.0",
+    "safe-memory-cache": "^2.0.0",
```

Comment:
Please check what's been done there. We've been using some internal property to handle some special cases.